File: CS/InteractiveNotifications/MainViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using DevExpress.Mvvm;

namespace InteractiveNotifications {
    public class MainViewModel {
        protected virtual INotificationService NotificationService => null;

        public void ShowNotification() {
            string text1 = "Lorem ipsum dolor sit amet integer fringilla, dui eget ultrices cursus, justo tellus.";
            string text2 = "In ornare ante magna, eget volutpat mi bibendum a. Nam ut ullamcorper libero. Pellentesque habitant.";
            string text3 = "Quisque sapien odio, mollis tincidunt est id, fringilla euismod neque. Aenean adipiscing lorem dui, nec. ";
            INotification notification = NotificationService.CreatePredefinedNotification(text1, text2, text3);
            Show(notification);
        }

        void Show(INotification notification) {
            CreateLogLine("Showing...");
            notification.ShowAsync().ContinueWith(OnNotificationShown, TaskScheduler.FromCurrentSynchronizationContext());
        }
        void OnNotificationShown(Task<NotificationResult> task) {
            try {
                switch(task.Result) {
                    case NotificationResult.Activated:
                        CreateLogLine("Activated");
                        break;
                    case NotificationResult.TimedOut:
                        CreateLogLine("Timed out");
                        break;
                    case NotificationResult.UserCanceled:
                        CreateLogLine("Canceled by user");
                        break;
                    case NotificationResult.Dropped:
                        CreateLogLine("Dropped (the queue is full)");
                        break;
                }
            } catch(AggregateException e) {
                CreateLogLine("Error: " + e.InnerException.Message);
            }
        }

        void CreateLogLine(string text) {
            System.Diagnostics.Debug.WriteLine(text);
        }
    }
}

File: VB/InteractiveNotifications/MainViewModel.vb
Imports System
Imports System.Threading.Tasks
Imports DevExpress.Mvvm

Namespace InteractiveNotifications

    Public Class MainViewModel

        Protected Overridable ReadOnly Property NotificationService As INotificationService
            Get
                Return Nothing
            End Get
        End Property

        Public Sub ShowNotification()
            Dim text1 As String = "Lorem ipsum dolor sit amet integer fringilla, dui eget ultrices cursus, justo tellus."
            Dim text2 As String = "In ornare ante magna, eget volutpat mi bibendum a. Nam ut ullamcorper libero. Pellentesque habitant."
            Dim text3 As String = "Quisque sapien odio, mollis tincidunt est id, fringilla euismod neque. Aenean adipiscing lorem dui, nec. "
            Dim notification As INotification = NotificationService.CreatePredefinedNotification(text1, text2, text3)
            Show(notification)
        End Sub

        Private Sub Show(ByVal notification As INotification)
            CreateLogLine("Showing...")
            notification.ShowAsync().ContinueWith(New Action(Of Task(Of NotificationResult))(AddressOf OnNotificationShown), TaskScheduler.FromCurrentSynchronizationContext())
        End Sub

        Private Sub OnNotificationShown(ByVal task As Task(Of NotificationResult))
            Try
                Select Case task.Result
                    Case NotificationResult.Activated
                        CreateLogLine("Activated")
                    Case NotificationResult.TimedOut
                        CreateLogLine("Timed out")
                    Case NotificationResult.UserCanceled
                        CreateLogLine("Canceled by user")
                    Case NotificationResult.Dropped
                        CreateLogLine("Dropped (the queue is full)")
                End Select
            Catch e As AggregateException
                CreateLogLine("Error: " & e.InnerException.Message)
            End Try
        End Sub

        Private Sub CreateLogLine(ByVal text As String)
            System.Diagnostics.Debug.WriteLine(text)
        End Sub
    End Class
End Namespace
